src/com/yetanalytics/lrs_admin_ui/views/status.cljs

```diff
@@ -0,0 +1,227 @@
+    [:div.vis-timeline-controls-select-unit
+     [:label
+      {:for "timeline-select-unit"}
+      "Time Unit"]
+     (into [:select
+            {:id "timeline-select-unit"
+             :value current-unit
+             :on-change #(dispatch [:status/set-timeline-unit
+                                    (fns/ps-event-val %)])}]
+           (for [unit ["year"
+                       "month"
+                       "day"
+                       "hour"
+                       "minute"
+                       "second"]]
+             [:option
+              {:value unit
+               :key (str "unit-" unit)}
+              (cs/capitalize unit)]))]))
+
+(defn- pick-datetime
+  [input-id
+   label
+   value
+   on-change
+   min-datetime
+   & [max-datetime]]
+  [:div.vis-timeline-controls-pick-datetime
+   [:label
+    {:for input-id}
+    label]
+   [:input
+    (cond-> {:id input-id
+             :type "datetime-local"
+             :value value
+             :on-change on-change
+             :min min-datetime}
+      max-datetime (assoc :max max-datetime))]])
+
+(defn timeline-pick-since
+  []
+  [pick-datetime
+   "timeline-pick-since"
+   "Since"
+   @(subscribe [:status.params/timeline-since-local])
+   #(dispatch [:status/set-timeline-since
+               (fns/ps-event-val %)])
+   "1970-01-01T00:00:00"
+   @(subscribe [:status.params/timeline-until-local])])
+
+(defn timeline-pick-until
+  []
+  [pick-datetime
+   "timeline-pick-until"
+   "Until"
+   @(subscribe [:status.params/timeline-until-local])
+   #(dispatch [:status/set-timeline-until
+               (fns/ps-event-val %)])
+   @(subscribe [:status.params/timeline-since-local])])
+
+(defn timeline-controls
+  []
+  [:div.vis-timeline-controls
+   [timeline-select-unit]
+   [timeline-pick-since]
+   [timeline-pick-until]])
+
+(def unit-for
+  "Mapping of timeline bucket time unit to suitable FOR sql substring arg."
+  {"year"   4
+   "month"  7
+   "day"    10
+   "hour"   13
+   "minute" 16
+   "second" 19})
+
+(defn- x-tick-format
+  [unit js-date]
+  (subs (.toISOString js-date) 0 (get unit-for unit)))
+
+(defn timeline-chart
+  []
+  (let [x-unit @(subscribe [:status.params/timeline-unit])]
+    [:div.vis-timeline-chart
+     [vis/chart
+      {:domain @(subscribe [:status.data.timeline/domain])
+       :min-domain {:y 0}
+       :container-component (r/as-element [vis/zoom
+                                           {:zoom-dimension "x"}])
+       :domain-padding 10
+       :theme (aget vis/theme "material")
+       :height 200
+       :padding {:top 10 :left 40 :right 30 :bottom 70}}
+      [vis/scatter
+       {:standalone false
+        :label-component (r/as-element
+                          [vis/tooltip
+                           {:style {:font-size 8}}])
+        :labels (fn [c]
+                  (let [datum (aget c "datum")
+                        x (aget datum "x")
+                        y (aget datum "y")]
```

Comment:
Could probably refactor out this twice-repeated datum code

Reply:
https://github.com/yetanalytics/lrs-admin-ui/pull/39/commits/e533aef3b10aa601191e827e349846984c5e080d